tern/formats/cyclonedx/cyclonedx_common.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: BSD-2-Clause
+
+"""
+Common functions that are useful for CycloneDX document creation
+"""
+
+import datetime
+import uuid
+from tern.utils import general
+
+
+###################
+# General Helpers #
+###################
+
+
+# document level tool information
+metadata_tool = {
+    'vendor': 'Tern Tools',
+    'name': 'Tern',
+    'version': general.get_git_rev_or_version()[1]
+}
+
+
+# keys are what Tern uses, values what CycloneDX uses
+hash_type_mapping = {
+    'md5': 'MD5',
+    'sha1': 'SHA-1',
+    'sha256': 'SHA-256',
+}
+
+
+pkg_format_purl_type_mapping = {
+    'deb': 'deb',
+    'rpm': 'rpm',
+    'apk': 'apk',
+    'pip': 'pip',
+    'gem': 'gem',
+    'npm': 'npm',
+    'go.mod': 'go',
+}
+
+
+purl_types_with_namespaces = [
+    'deb',
+    'rpm',
+    'apk',
+]
+
+
+# map Tern OS guesses to package URL namespace
+os_guess_purl_namespace_mapping = {
```

Comment:
I wonder if there is an easier way to figure out a valid global namespace for all the ecosystems. Does CycloneDX require a specific string format for namespace or can we just do something like replace spaces with underscores?

Reply:
This is determined by the package URL spec. This might have been me over thinking future OS support. I'll refactor this part and we can always come back in the future and implement an override mapping for any exceptions instead.